src/class-wpml-gutenberg-integration-factory.php

```diff
@@ -1,29 +1,55 @@
 <?php
 
 use function WPML\Container\make;
+use function WPML\Container\share;
 
 class WPML_Gutenberg_Integration_Factory {
 
-	/** @return \WPML\PB\Gutenberg\Integration */
+	/** @return \WPML\PB\Gutenberg\Integration_Composite */
 	public function create() {
-		/**
-		 * @var SitePress $sitepress
-		 * @var wpdb      $wpdb
-		 */
-		global $sitepress, $wpdb;
+		$integrations = new WPML\PB\Gutenberg\Integration_Composite();
 
-		$config_option = new WPML_Gutenberg_Config_Option();
+		$mainIntegration = $this->create_gutenberg_integration();
+		share( [ $mainIntegration ] );
```

Comment:
By sharing this instance of the main integration class, we can reuse it in the other classes by just `make( \WPML_Gutenberg_Integration::class );`.